pkg/virt-launcher/virtwrap/network/podnic.go

```diff
@@ -194,12 +198,20 @@ func (l *podNIC) PlugPhase1() error {
 	if dhcpConfig != nil {
 		log.Log.V(4).Infof("The generated dhcpConfig: %s", dhcpConfig.String())
 		if err := l.cacheFactory.CacheDHCPConfigForPid(getPIDString(l.launcherPID)).Write(l.podInterfaceName, dhcpConfig); err != nil {
-			log.Log.Reason(err).Error("failed to save dhcpConfig configuration")
-			return errors.CreateCriticalNetworkError(err)
+			return fmt.Errorf("failed to save DHCP configuration: %w", err)
 		}
 	}
-
 	domainIface := l.infraConfigurator.GenerateDomainIfaceSpec()
```

Comment:
I would change `GenerateDomainIfaceSpec` to `GenerateNonRecoverableDomainIfaceSpec` now it's the only purpose of this method.

Reply:
I find non-recoverable confusing - could you elaborate why you want this distinction ? Is there a recoverable domain interface spec being retrieved elsewhere ?

Reply:
This is not part of this PR.